os/kernel/binary_manager/binary_manager_recovery.c

```diff
@@ -68,62 +63,51 @@ static void binary_manager_board_reset(void)
 #endif
 }
 
-static void recovery_kill_each(FAR struct tcb_s *tcb, FAR void *arg)
+static void recovery_exclude_scheduling_each(FAR struct tcb_s *tcb, FAR void *arg)
 {
-	int ret;
-	binmgr_kill_data_t *info;
+	int binid;
+	irqstate_t flags;
 
-	info = (binmgr_kill_data_t *)arg;
-	if (info == NULL || info->binid < 0 || info->faultid < 0) {
+	binid = (int)arg;
+	if (binid < 0) {
 		return;
 	}
 
-	if (tcb->group->tg_loadtask == info->binid && tcb->pid != info->binid) {
-		bmllvdbg("KILL!! %d\n", tcb->pid);
-		ret = task_terminate(tcb->pid, true);
+	if (tcb->group->tg_loadtask == binid) {
+		flags = irqsave();
+		(void)sched_removereadytorun(tcb);
+		sched_addblocked(tcb, TSTATE_TASK_INACTIVE);
+		irqrestore(flags);
+		bmllvdbg("Remove pid %d from readytorun list\n", tcb->pid);
 	}
 }
 
 /****************************************************************************
- * Name: recovery_kill_binary
+ * Name: recovery_exclude_scheduling
  *
  * Description:
- *   This function will kill all the task/thread created by the binary
- *   i.e input pid.
- *   If any fault occurs in the system, recovery thread will call this function to
- *   kill all the thread/task created by the input binary.
+ *   This function will remove all the task/thread created by the binary
+ *   i.e input pid from ready to run list.
  *
  * Input parameters:
- *   pid   -   The pid of the binary, whoes all childs to be killed
+ *   pid   -   The pid of the binary, whoes all children to be killed
  *
  * Returned Value:
  *   Zero (OK) on success; otherwise -1 (ERROR) value is returned.
  *
  ****************************************************************************/
-static int recovery_kill_binary(pid_t pid, int binid)
+static int recovery_exclude_scheduling(int binid)
 {
 	int ret;
+	irqstate_t flags;
+	FAR struct tcb_s *tcb;
 
-	binmgr_kill_data_t info;
-
-	if (pid < 0 || binid < 0) {
+	if (binid < 0) {
 		return ERROR;
 	}
 
-	/* Set binary id and faulty id */
-	info.binid = binid;
-	info.faultid = pid;
-
-	/* Search all task/pthreads created by same loaded task */
-	sched_foreach(recovery_kill_each, (FAR void *)&info);
-
-	/* Finally, unload binary */
-	ret = task_terminate(binid, true);
-	if (ret < 0) {
-		bmlldbg("Failed to unload binary %d, ret %d, errno %d\n", binid, ret, errno);
-		return ERROR;
-	}
-	bmllvdbg("Unload binary! pid %d\n", binid);
+	/* Search all task/pthreads created by same loaded task and remove them from ready to run list */
```

Comment:
How about below?
Remove all tasks and pthreads created in a binary which has "binid" from readytorun list